### PR TITLE
feat(schemas): add date format, offset, and amount validation

### LINKED ISSUE #111: feat(schemas): add date format, offset, and amount validation to Zod schemas
## Summary

Zod input schemas lack client-side validation for date formats, pagination offsets, and monetary amounts. Invalid values pass through to the freee API, resulting in poor error messages.

## Problem

**File:** `src/schemas.ts`

### Date fields — no format validation
```typescript
startIssueDate: z.string().optional().describe('Start date (YYYY-MM-DD)'),
endIssueDate: z.string().optional().describe('End date (YYYY-MM-DD)'),
```
Accepts any string: `"not-a-date"`, `"2024-13-45"`, etc.

### Offset — negative values allowed
```typescript
offset: z.number().optional().describe('Pagination offset'),
```

### Amount — zero/negative values allowed in some schemas
Some schemas like `CreateManualJournalSchema` correctly use `.min(1)`, but others (e.g., `CreateTransferSchema`, `CreateDealPaymentSchema`) have no minimum constraint on amount fields.

## Impact

- Invalid dates reach freee API and return cryptic error messages
- Negative offsets cause undefined behavior
- Zero/negative amounts for financial transactions are invalid but accepted

## Acceptance Criteria

- [ ] Create reusable `dateField` helper: `z.string().regex(/^\d{4}-\d{2}-\d{2}$/, 'Date must be YYYY-MM-DD format')` (following existing `companyIdField` pattern)
- [ ] Apply date validation to all date fields across all schemas
- [ ] Add `.min(0)` to all `offset` fields
- [ ] Add `.min(1)` or `.positive()` to all monetary `amount` fields (align with `CreateManualJournalSchema` pattern)
- [ ] Add `limit` field max constraint (freee API max is typically 100)
- [ ] Add unit tests for validation edge cases

## Files to Modify

- `src/schemas.ts` (primary)
- `src/__tests__/schemas.test.ts` (add validation tests)

### PR DESCRIPTION
**Pull request by knishioka**
## Summary

- Create reusable `dateField`/`optionalDateField` helpers with `YYYY-MM-DD` regex validation, following existing `companyIdField` pattern
- Apply date validation to all 30 date fields across 16 schemas
- Add `.min(0)` to all 9 `offset` fields for pagination safety
- Add `.min(1)` to 4 monetary `amount` fields (aligning with `CreateManualJournalSchema` pattern)
- Add 121 unit tests covering validation edge cases

Closes #111

## Changes

| File | Change |
|---|---|
| `src/schemas.ts` | Add helpers, apply validations (60 insertions, 52 deletions) |
| `src/__tests__/schemas.test.ts` | New: 121 validation tests (603 lines) |

## Validation Details

**Date fields** (30 total): Regex `/^\d{4}-\d{2}-\d{2}$/` ensures YYYY-MM-DD format before API call. 7 required + 23 optional fields.

**Offset fields** (9 total): `.min(0)` prevents negative pagination offsets.

**Amount fields** (4 new + 1 existing): `.min(1)` on monetary transaction amounts (`CreateDealSchema`, `UpdateDealSchema`, `CreateDealPaymentSchema`, `CreateTransferSchema`). Filter fields (`minAmount`/`maxAmount`) intentionally excluded.

## Test plan

- [x] Invalid date formats rejected: `"not-a-date"`, `"2024/01/01"`, `""`, `"2024-1-1"`
- [x] Valid date format `"2024-01-15"` accepted
- [x] Optional date fields accept `undefined`
- [x] Negative offsets (`-1`) rejected, `0` accepted
- [x] Zero/negative amounts rejected, `1` accepted
- [x] Full integration tests with realistic payloads
- [x] All 455 tests pass (121 new + 334 existing)
- [x] Lint, typecheck, and build all pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/__tests__/schemas.test.ts [hunk 1]
```diff
@@ -0,0 +1,603 @@
...
+  });
+
+  it('should reject out-of-range date "2024-13-45" (regex only, not calendar validation)', () => {
+    // The regex /^\d{4}-\d{2}-\d{2}$/ matches the format but does not validate
+    // actual month/day ranges. "2024-13-45" matches \d{4}-\d{2}-\d{2}.
+    // This test documents the current behavior: regex-only validation passes
+    // for syntactically correct but semantically invalid dates.
+    const result = schema.safeParse({ issueDate: '2024-13-45' });
+    expect(result.success).toBe(true);
+  });
```
[the remark this answers is not in the document]
Comment (knishioka) [reply src/__tests__/schemas.test.ts]: Fixed in commit 20dd639. Renamed test to 'should reject semantically invalid date' and updated assertion to `toBe(false)` now that calendar validation is in place. Also added tests for impossible dates (2024-02-30) and leap year validation.
